src/components/EditionModal/EditionModal.jsx

```diff
@@ -0,0 +1,167 @@
+  category: 'category',
+  accountOrGroup: 'accountOrGroup',
+  threshold: 'threshold'
+}
+
+const SectionsPerType = {
+  [CHOOSING_TYPES.category]: CategorySection,
+  [CHOOSING_TYPES.accountOrGroup]: AccountOrGroupSection,
+  [CHOOSING_TYPES.threshold]: ThresholdSection
+}
+
+const ChoosingSwitch = ({ choosing }) => {
+  return (
+    <>
+      {choosing.type === CHOOSING_TYPES.category ? (
+        <CategoryChoice
+          modal={false}
+          canSelectParent={true}
+          categoryId={choosing.value.id}
+          categoryIsParent={choosing.value.isParent}
+          onSelect={choosing.onSelect}
+          onCancel={choosing.onCancel}
+        />
+      ) : null}
+      {choosing.type === CHOOSING_TYPES.accountOrGroup ? (
+        <AccountGroupChoice
+          current={choosing.value}
+          onSelect={choosing.onSelect}
+          onCancel={choosing.onCancel}
+        />
+      ) : null}
+    </>
+  )
+}
+
+const InfoSlide = ({
+  doc,
+  fieldSpecs,
+  fieldOrder,
+  fieldLabels,
+  onRequestChooseField,
+  onChangeField
+}) => {
+  return (
+    <ModalSections>
+      {fieldOrder.map(fieldName => {
+        const FieldSection = SectionsPerType[fieldSpecs[fieldName].type]
+        const fieldSpec = fieldSpecs[fieldName]
+        const fieldLabel = fieldLabels[fieldName]
+        return (
+          <FieldSection
+            key={fieldName}
+            label={fieldLabel}
+            value={fieldSpec.getValue(doc)}
+            onClick={
+              fieldSpec.immediate ? null : () => onRequestChooseField(fieldName)
+            }
+            onChange={
+              fieldSpec.immediate
+                ? ev => {
+                    onChangeField(fieldName, ev.target.value)
+                  }
+                : null
+            }
+          />
+        )
+      })}
+    </ModalSections>
+  )
+}
+
+const EditionModal = props => {
+  const {
+    fieldSpecs,
+    fieldLabels,
+    fieldOrder,
+    initialDoc,
+    modalTitle,
+    okButtonLabel,
+    cancelButtonLabel,
+    onEdit,
+    onDismiss
+  } = props
+  const [doc, setDoc] = useState(initialDoc)
+  const [choosing, setChoosing] = useState(null)
+
+  const handleChoosingCancel = () => {
+    setChoosing(null)
+  }
+
+  const handleChangeField = (name, val) => {
+    const updater = fieldSpecs[name].updater
+    const updatedDoc = updater(doc, val)
+    setDoc(updatedDoc)
+  }
+
+  const handleRequestChooseField = name => {
+    const fieldSpec = fieldSpecs[name]
+    setChoosing({
+      type: fieldSpec.type,
+      value: fieldSpec.getValue(doc),
+      onSelect: val => {
+        setChoosing(null)
+        handleChangeField(name, val)
+      },
+      onCancel: handleChoosingCancel
+    })
+  }
+
+  const handleConfirmEdit = () => {
+    onEdit(doc)
+  }
+
+  return (
+    <Modal title={modalTitle} mobileFullscreen={true} dismissAction={onDismiss}>
+      <Stepper
+        showPercentage={false}
+        currentIndex={choosing ? 1 : 0}
```

Comment:
Nit: Even if they look self explanatory at first sight, these 1 and 0 look a lot like magic values. I'm afraid we will not be comfortable with it in the future. Maybe we should put it in variables with clear names or add a comment?

Reply:
Good points, [solved](https://github.com/cozy/cozy-banks/pull/1706/commits/d6d7d6828e8ec4d331fac39018cc4be68c10b74d)